### PR TITLE
Creation of CRUD for users and operations for add and remove books

### PR DESCRIPTION
**Pull request by JJhoan**
## Summary
-Added Crud to user
-Added methods for add and remove books

##Trello Card
https://trello.com/c/gbgKcodC/21-crear-user-controller-con-las-operaciones-crud-de-user

### CODE REVIEW

File: src/main/java/com/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(UserController.USERS)
+public class UserController {
+
+    public static final String USERS = "/api/users";
+    public static final String ADD_BOOK = "/addBook/{idUser}/{idBook}";
+    public static final String REMOVE_BOOK = "/removeBook/{idUser}/{idBook}";
```
Comment (jeymarsonwolox) [inline src/main/java/com/wolox/training/controllers/UserController.java]: you can extract all routes (books and users) in one file

File: src/main/java/com/wolox/training/controllers/UserController.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        if (!userRepository.findById(user.getIdUser()).isPresent()) {
+            throw new UserNotFoundException("User not found");
+        }
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping(value = "/{idUser}")
+    public void deleteUser(@PathVariable Long idUser) {
+        if (!userRepository.findById(idUser).isPresent()) {
+            throw new UserNotFoundException("User not found");
```
Comment (jeymarsonwolox) [inline src/main/java/com/wolox/training/controllers/UserController.java]: you repeat `User not found` extract to a constants

File: src/main/java/com/wolox/training/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        }
+        userRepository.deleteById(idUser);
+    }
+
+    @PutMapping (value = ADD_BOOK)
+    public void addBook(@PathVariable Long idUser, @PathVariable Long idBook) {
+        Book book = bookRepository.findById(idBook).orElseThrow(() -> new BookNotFoundException("Book not found"));
```
Comment (jeymarsonwolox) [inline src/main/java/com/wolox/training/controllers/UserController.java]: extract `Book not found` to a constants

**Review by jeymarsonwolox — CHANGES_REQUESTED** [review]

**Review by jeymarsonwolox — DISMISSED** [review]
Excellent!!

**Review by jeymarsonwolox — DISMISSED** [review]

File: src/main/java/com/wolox/training/constants/Message.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    public static final String INVALID_INVOCATION = "Invalid invocation";
+    public static final String BOOK_ALREADY_EXIST = "Book already exists.";
+    public static final String BOOK_NOT_FOUND = "Book not found";
+    public static final String USER_NOT_FOUND = "User not found";
+    public static final String USER_ALREADY_EXIST = "User already exists.";
+    public static final String NOT_FOUND = "Not found";
+    public static final String USER_HAS_NOT_BOOK = "User has not the book";
+    public static final String BOOK_CAN_NOT_NULL = "Book can not be null";
+    public static final String BOOK_IS_ALREADY_ASSOCIATED = "Book is already associated";
+    public static final String BOOK_CAN_BE_NULL = "Book can not be null";
```
Comment (IgnOliveto) [inline src/main/java/com/wolox/training/constants/Message.java]: The enum here should be "BOOK_CANT_BE_NULL"

**Review by jeymarsonwolox — DISMISSED** [review]

**Review by IgnOliveto — APPROVED** [review]

**Review by jeymarsonwolox — APPROVED** [review]

**Pull request merged**